lib/wallet/wallet.js

```diff
@@ -1111,7 +1111,10 @@ class Wallet extends EventEmitter {
       options = {};
 
     const acct = options.account || 0;
-    const change = await this.changeAddress(acct);
+
+    let change = options.changeAddress;
+    if(change == null)
+      change = await this.changeAddress(acct);
 
     if (!change)
       throw new Error('Account not found.');
```

Comment:
I think this "Account not found" check needs to be exclusively part of the path where a `changeAddress` was not provided. Maybe like this:

```diff
diff --git a/lib/wallet/wallet.js b/lib/wallet/wallet.js
index 301ad8de..51895d30 100644
--- a/lib/wallet/wallet.js
+++ b/lib/wallet/wallet.js
@@ -1113,11 +1113,12 @@ class Wallet extends EventEmitter {
     const acct = options.account || 0;

     let change = options.changeAddress;
-    if(change == null)
+    if (change == null) {
       change = await this.changeAddress(acct);

-    if (!change)
-      throw new Error('Account not found.');
+      if (!change)
+        throw new Error('Account not found.');
+    }

     let rate = options.rate;
     if (rate == null)
```